### DISCUSSION TITLE
How to get runtime symbols when building runtime ?

### DISCUSSION THREAD

**Discussion by jpdurot** [dotnet/runtime#109578] [Q&A] [answered]
Hi, when building the runtime in Release mode, we used to have (at least for .Net 6) a `dotnet-runtime-symbols-<platform>-<version>.tar.gz` archive in the Shipping directory, containing all the PDB and R2R files needed to debug the custom runtime. It looks like we don't have any generated symbols for .Net 8 (and .Net 9). Is there any flag we could add to the build command line to generate symbols in Release configuration ?

**Comment by hez2010** [comment]
You can try generating the core_root layout after building clr+libs using 
```
./src/tests/build.sh Release generatelayoutonly
```
Then all necessary files including coreclr, libraries and their pdb will be placed under `artifacts/tests/coreclr/<platform>/Tests/Core_Root`

**Comment by am11** [comment, marked as answer]
It was changed in https://github.com/dotnet/runtime/pull/90050. Besides "use source-build" which has a slightly different workflow, there are two options:
1. override `SkipArchivesBuild` from build command `./build.sh clr+libs+packs -c Release -p:SkipArchivesBuild=false`; it produces `dotnet-runtime-symbols-<platform>-<version>.tar.gz` as before.
2. disable symbol stripping (to separate files) `./build.sh clr+libs+packs -keepnativesymbols true -c Release`; it produces fat/unstripped native binaries containing the debug symbols within.

**Reply by jpdurot** [reply]
Thanks a lot @am11, I tried the first option and indeed I now get the symbol archive !
